protostar/testing/test_collector.py

```diff
@@ -275,12 +277,25 @@ def _build_test_suite_from_test_suite_info(
         test_suite_info: TestSuiteInfo,
     ) -> TestSuite:
         function_names = self._get_suite_function_names(test_suite_info.path)
-        setup_fn_name = self._collect_setup_hook_name(function_names)
+
+        if len(function_names) == 0:
+            names: List[str] = []
+            gas: List[AvailableGas] = []
+        elif isinstance(function_names[0], str):
+            function_names = cast(List[str], function_names)
+            names: List[str] = function_names
+            gas: List[AvailableGas] = [None] * len(function_names)
+        else:
```

Comment:
we should still check for instance here and throw instead of casting

Reply:
type checker screams that checking for instance here is unnecessary, yet it still needs the cast to determine the type